### PR TITLE
[Bugfix] Add random_seed to sample_hf_requests in benchmark_serving script

### PR DESCRIPTION
**Pull request by wukaixingxp**
[Bugfix] Current sample_hf_requests() in benchmark_serving.py did not have a good reproducibility as it uses `dataset.shuffle()`, which causes the input prompts to be different between each run. 

This PR adds `random_seed=arg.seed` argument to sample_hf_requests() and changes `dataset.shuffle()` to `dataset.shuffle(seed=RANDOM_SEED)` to ensure benchmark reproducibility.  After this fix, the `Total input tokens` is now fixed to 12428 as shown in the test logs below.

**Test Log**:
first run:
```
python ../benchmark_serving.py --backend openai-chat --base-url http://0.0.0.0:8000/v1 --endpoint /chat/completions --model meta-llama/Llama-3.2-11B-Vision-Instruct --dataset-path lmms-lab/LLaVA-OneVision-Data --dataset-name hf --hf-subset "chart2text(cauldron)" --hf-split train --save-result
Namespace(backend='openai-chat', base_url='http://0.0.0.0:8000/v1', host='localhost', port=8000, endpoint='/chat/completions', dataset=None, dataset_name='hf', dataset_path='lmms-lab/LLaVA-OneVision-Data', model='meta-llama/Llama-3.2-11B-Vision-Instruct', tokenizer=None, best_of=1, use_beam_search=False, num_prompts=1000, logprobs=None, request_rate=inf, seed=0, trust_remote_code=False, disable_tqdm=False, profile=False, save_result=True, metadata=None, result_dir=None, result_filename=None, percentile_metrics='ttft,tpot,itl', metric_percentiles='99', sonnet_input_len=550, sonnet_output_len=150, sonnet_prefix_len=200, sharegpt_output_len=None, random_input_len=1024, random_output_len=128, random_range_ratio=1.0, random_prefix_len=0, hf_subset='chart2text(cauldron)', hf_split='train', hf_output_len=None)
Downloading readme: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 46.3k/46.3k [00:00<00:00, 9.22MB/s]
Starting initial single prompt test run...
Initial test run completed. Starting main benchmark run...
Traffic request rate: inf
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1000/1000 [03:35<00:00,  4.63it/s]
============ Serving Benchmark Result ============
Successful requests:                     1000      
Benchmark duration (s):                  215.79    
Total input tokens:                      12428     
Total generated tokens:                  63701     
Request throughput (req/s):              4.63      
Output token throughput (tok/s):         295.20    
Total Token throughput (tok/s):          352.79    
---------------Time to First Token----------------
Mean TTFT (ms):                          107762.34 
Median TTFT (ms):                        107688.53 
P99 TTFT (ms):                           210873.42 
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          101.65    
Median TPOT (ms):                        103.52    
P99 TPOT (ms):                           126.47    
---------------Inter-token Latency----------------
Mean ITL (ms):                           101.39    
Median ITL (ms):                         21.78     
P99 ITL (ms):                            480.65    
==================================================
```
second run:
```
python ../benchmark_serving.py --backend openai-chat --base-url http://0.0.0.0:8000/v1 --endpoint /chat/completions --model meta-llama/Llama-3.2-11B-Vision-Instruct --dataset-path lmms-lab/LLaVA-OneVision-Data --dataset-name hf --hf-subset "chart2text(cauldron)" --hf-split train --save-result
Namespace(backend='openai-chat', base_url='http://0.0.0.0:8000/v1', host='localhost', port=8000, endpoint='/chat/completions', dataset=None, dataset_name='hf', dataset_path='lmms-lab/LLaVA-OneVision-Data', model='meta-llama/Llama-3.2-11B-Vision-Instruct', tokenizer=None, best_of=1, use_beam_search=False, num_prompts=1000, logprobs=None, request_rate=inf, seed=0, trust_remote_code=False, disable_tqdm=False, profile=False, save_result=True, metadata=None, result_dir=None, result_filename=None, percentile_metrics='ttft,tpot,itl', metric_percentiles='99', sonnet_input_len=550, sonnet_output_len=150, sonnet_prefix_len=200, sharegpt_output_len=None, random_input_len=1024, random_output_len=128, random_range_ratio=1.0, random_prefix_len=0, hf_subset='chart2text(cauldron)', hf_split='train', hf_output_len=None)
Starting initial single prompt test run...
Initial test run completed. Starting main benchmark run...
Traffic request rate: inf
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1000/1000 [03:43<00:00,  4.47it/s]
============ Serving Benchmark Result ============
Successful requests:                     1000      
Benchmark duration (s):                  223.62    
Total input tokens:                      12428     
Total generated tokens:                  63680     
Request throughput (req/s):              4.47      
Output token throughput (tok/s):         284.77    
Total Token throughput (tok/s):          340.35    
---------------Time to First Token----------------
Mean TTFT (ms):                          112154.70 
Median TTFT (ms):                        111527.32 
P99 TTFT (ms):                           219165.63 
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          105.54    
Median TPOT (ms):                        106.54    
P99 TPOT (ms):                           136.76    
---------------Inter-token Latency----------------
Mean ITL (ms):                           105.57    
Median ITL (ms):                         22.22     
P99 ITL (ms):                            489.19    
==================================================
```
**BEFORE SUBMITTING, PLEASE READ THE CHECKLIST BELOW AND FILL IN THE DESCRIPTION ABOVE**

---

<details>

<summary><b> PR Checklist (Click to Expand) </b></summary>

<p>Thank you for your contribution to vLLM! Before submitting the pull request, please ensure the PR meets the following criteria. This helps vLLM maintain the code quality and improve the efficiency of the review process.</p>

<h3>PR Title and Classification</h3>
<p>Only specific types of PRs will be reviewed. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
<ul>
    <li><code>[Bugfix]</code> for bug fixes.</li>
    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
    <li><code>[Frontend]</code> For changes on the vLLM frontend (e.g., OpenAI API server, <code>LLM</code> class, etc.) </li>
    <li><code>[Kernel]</code> for changes affecting CUDA kernels or other compute kernels.</li>
    <li><code>[Core]</code> for changes in the core vLLM logic (e.g., <code>LLMEngine</code>, <code>AsyncLLMEngine</code>, <code>Scheduler</code>, etc.)</li>
    <li><code>[Hardware][Vendor]</code> for hardware-specific changes. Vendor name should appear in the prefix (e.g., <code>[Hardware][AMD]</code>).</li>
    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
</ul>
<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>

<h3>Code Quality</h3>

<p>The PR need to meet the following code quality standards:</p>

<ul>
    <li>We adhere to <a href="https://google.github.io/styleguide/pyguide.html">Google Python style guide</a> and <a href="https://google.github.io/styleguide/cppguide.html">Google C++ style guide</a>.</li>
    <li>Pass all linter checks. Please use <a href="https://github.com/vllm-project/vllm/blob/main/format.sh"><code>format.sh</code></a> to format your code.</li>
    <li>The code need to be well-documented to ensure future contributors can easily understand the code.</li>
    <li>Include sufficient tests to ensure the project to stay correct and robust. This includes both unit tests and integration tests.</li>
    <li>Please add documentation to <code>docs/source/</code> if the PR modifies the user-facing behaviors of vLLM. It helps vLLM user understand and utilize the new features or changes.</li>
</ul>

<h3>Adding or changing kernels</h3>
<p>Each custom kernel needs a schema and one or more implementations to be registered with PyTorch.</p>
<ul>
    <li>Make sure custom ops are registered following PyTorch guidelines: <a href="https://pytorch.org/tutorials/advanced/cpp_custom_ops.html#cpp-custom-ops-tutorial">Custom C++ and CUDA Operators</a> and <a href="https://docs.google.com/document/d/1_W62p8WJOQQUzPsJYa7s701JXt0qf2OfLub2sbkHOaU">The Custom Operators Manual</a></li>
    <li>Custom operations that return <code>Tensors</code> require meta-functions. Meta-functions should be implemented and registered in python so that dynamic dims can be handled automatically. See above documents for a description of meta-functions.</li>
    <li>Use <a href="https://pytorch.org/docs/stable/library.html#torch.library.opcheck"><code>torch.libary.opcheck()</code></a> to test the function registration and meta-function for any registered ops.  See <code>tests/kernels</code> for examples.</li>
    <li>When changing the C++ signature of an existing op, the schema must be updated to reflect the changes.</li>
    <li>If a new custom type is needed, see the following document: <a href="https://docs.google.com/document/d/18fBMPuOJ0fY5ZQ6YyrHUppw9FA332CpNtgB6SOIgyuA">Custom Class Support in PT2</a>.
</ul>

<h3>Notes for Large Changes</h3>
<p>Please keep the changes as concise as possible. For major architectural changes (>500 LOC excluding kernel/data/config/test), we would expect a GitHub issue (RFC) discussing the technical design and justification. Otherwise, we will tag it with <code>rfc-required</code> and might not go through the PR.</p>

<h3>What to Expect for the Reviews</h3>

<p>The goal of the vLLM team is to be a <i>transparent reviewing machine</i>. We would like to make the review process transparent and efficient and make sure no contributor feel confused or frustrated. However, the vLLM team is small, so we need to prioritize some PRs over others. Here is what you can expect from the review process: </p>

<ul>
    <li> After the PR is submitted, the PR will be assigned to a reviewer. Every reviewer will pick up the PRs based on their expertise and availability.</li>
    <li> After the PR is assigned, the reviewer will provide status update every 2-3 days. If the PR is not reviewed within 7 days, please feel free to ping the reviewer or the vLLM team.</li>
    <li> After the review, the reviewer will put an <code> action-required</code> label on the PR if there are changes required. The contributor should address the comments and ping the reviewer to re-review the PR.</li>
    <li> Please respond to all comments within a reasonable time frame. If a comment isn't clear or you disagree with a suggestion, feel free to ask for clarification or discuss the suggestion.
 </li>
</ul>

<h3>Thank You</h3>

<p> Finally, thank you for taking the time to read these guidelines and for your interest in contributing to vLLM. Your contributions make vLLM a great tool for everyone! </p>

</details>

### CODE REVIEW

**Review by Isotr0py — APPROVED** [review]
Sorry for the delay. Thanks for fixing this!

Can you please fix the lint errors through running `bash format.sh`? Thanks!

**Comment by wukaixingxp** [comment]
@Isotr0py Thanks for your review. I run ` format.sh` and did not see any changed made by `format.sh`. I do not think my file has any lint error.
```
(vllm) [ ~/work/kai/vllm (main)]$ bash format.sh
vLLM yapf: Done
vLLM mypy:
Running mypy on 
Success: no issues found in 128 source files
Running mypy on tests
Success: no issues found in 319 source files
Running mypy on vllm/assets
Success: no issues found in 5 source files
Running mypy on vllm/attention
Success: no issues found in 16 source files
Running mypy on vllm/distributed
Success: no issues found in 12 source files
Running mypy on vllm/engine
Success: no issues found in 17 source files
Running mypy on vllm/entrypoints
Success: no issues found in 24 source files
Running mypy on vllm/executor
Success: no issues found in 17 source files
Running mypy on vllm/logging
Success: no issues found in 2 source files
Running mypy on vllm/lora
Success: no issues found in 17 source files
Running mypy on vllm/model_executor
Success: no issues found in 88 source files
Running mypy on vllm/multimodal
Success: no issues found in 7 source files
Running mypy on vllm/platforms
Success: no issues found in 7 source files
Running mypy on vllm/plugins
Success: no issues found in 1 source file
Running mypy on vllm/prompt_adapter
Success: no issues found in 6 source files
Running mypy on vllm/spec_decode
Success: no issues found in 16 source files
Running mypy on vllm/transformers_utils
Success: no issues found in 30 source files
Running mypy on vllm/usage
Success: no issues found in 2 source files
Running mypy on vllm/worker
Success: no issues found in 23 source files
vLLM mypy: Done
vLLM codespell: Done
vLLM ruff: Done
vLLM isort: Done
vLLM clang-format: Done
vLLM actionlint:
vLLM actionlint: Done
(vllm) [ ~/work/kai/vllm (main)]$ git add .
(vllm) [ ~/work/kai/vllm (main)]$ git commit -m 'formated by format.sh'
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean
(vllm) [~/work/kai/vllm (main)]$ git log --raw
commit 50e093943da564a32d86b37d399a26b66b8bd83b (HEAD -> main, origin/main, origin/HEAD)
Merge: dc1cf2fb 5d264f4a
Author: Kai Wu <wukaixingxp@gmail.com>
Date:   Tue Oct 15 14:27:23 2024 -0700

    Merge branch 'vllm-project:main' into main

commit 5d264f4ab8d008f0ac5b7f0adb7189d70136f3ec
Author: Grace Ho <146482179+gracehonv@users.noreply.github.com>
Date:   Tue Oct 15 13:30:44 2024 -0700

    pass ignore_eos parameter to all benchmark_serving calls (#9349)

:100644 100644 04999518 c1a396c8 M      benchmarks/benchmark_serving.py

commit e9d517f27673ec8736c026f2311d3c250d5f9061
Author: Nick Hill <nickhill@us.ibm.com>
Date:   Tue Oct 15 07:19:48 2024 +0100

    [BugFix] Fix chat API continuous usage stats (#9357)

:100644 100644 0fbc4cca 3af0032f M      tests/entrypoints/openai/test_chat.py
:100644 100644 9470b6ea acb56e4a M      vllm/entrypoints/openai/serving_chat.py

commit 55e081fbad29c6710318e1715372cc927e44de8b
```

**Comment by Isotr0py** [comment]
I see, you can refer to the output of the CI here: https://github.com/vllm-project/vllm/actions/runs/11354659097/job/31582376192?pr=9013#step:5:18

File: benchmarks/benchmark_serving.py [hunk 1]
```diff
@@ -210,7 +211,7 @@ def sample_hf_requests(
                            streaming=True)
     assert "conversations" in dataset.features, (
         "HF Dataset must have 'conversations' column.")
-    filtered_dataset = dataset.shuffle().filter(
+    filtered_dataset = dataset.shuffle(seed=random_seed).filter(
         lambda x: len(x["conversations"]) >= 2)
```
Comment (Isotr0py) [inline benchmarks/benchmark_serving.py:215]:
```suggestion
    filter_func = lambda x: len(x["conversations"]) >= 2
    filtered_dataset = dataset.shuffle(seed=random_seed).filter(filter_func)
```
@wukaixingxp This modification should fix the lint errors.
